src/smt/env.h

```diff
@@ -279,6 +280,26 @@ class Env
   /** get oracle checker */
   theory::quantifiers::OracleChecker* getOracleChecker() const;
 
+  /**
+   * Register Boolean term skolem. This registers that k is a Boolean variable
+   * that should be treated as a theory atom. This impacts theoryOf, where
+   * Boolean term skolems belong to THEORY_UF, not THEORY_BOOL.
+   *
+   * This method is called by the "remove term formula" pass, which recognizes
+   * when Boolean terms occur in term positions, which are relevant for
+   * theory combination.
```

Comment:
Not sure if this should be here, as this function can technically be used in other places too.

Reply:
I think it is helpful to state the main use case, but indeed the method can be used in other places.